portlet-mvc-util/src/main/java/org/jasig/portlet/spring/SpringELProcessor.java

```diff
@@ -0,0 +1,139 @@
+
+
+/**
+ * Processor that uses spring EL for the implementation.
+ *
+ * @author Josh Helmer, jhelmer@unicon.net
+ */
+@Service
+public class SpringELProcessor implements IExpressionProcessor, BeanFactoryAware {
+    private static final ParserContext PARSER_CONTEXT = new TemplateParserContext("${", "}");
+
+    protected Logger logger = LoggerFactory.getLogger(getClass());
+
+    private BeanResolver beanResolver;
+    private Properties properties;
+
+
+    /**
+     * @param beanFactory
+     * @{inheritDoc}
+     */
+    @Override
+    public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
+        this.beanResolver = new BeanFactoryResolver(beanFactory);
+    }
+
+
+    /**
+     * Set the properties loader to user.
+     *
+     * @param properties the properties loader
+     */
+    public void setProperties(final Properties properties) {
+        this.properties = properties;
+    }
+
+
+    /**
+     * @{inheritDoc}
+     */
+    @Override
+    public String process(String value, PortletRequest request) {
+        Map<String, Object> context = getContext(request);
+
+        StandardEvaluationContext sec = new StandardEvaluationContext(context);
+        sec.addPropertyAccessor(new MapAccessor());
+        sec.addPropertyAccessor(new ReflectivePropertyAccessor());
+        sec.addPropertyAccessor(new DefaultPropertyAccessor(
+                PARSER_CONTEXT.getExpressionPrefix(),
+                PARSER_CONTEXT.getExpressionSuffix()));
+        sec.setBeanResolver(beanResolver);
+        SpelExpressionParser parser = new SpelExpressionParser();
+
+        String processed = parser
+                .parseExpression(value, PARSER_CONTEXT)
+                .getValue(sec, String.class);
+
+        return processed;
+    }
+
+
+    /**
+     * Setup the context for spring EL.   Will add all raw properties from
+     * the config file, the request parameters and user info.
+     *
+     * @param request the portlet request to read params from
+     * @return a map of properties
+     */
+    private Map<String, Object> getContext(PortletRequest request) {
+        Map<String, Object> context = new HashMap<String, Object>();
+
+        if(properties != null) {
+            for (String key : properties.stringPropertyNames()) {
+                context.put(key, properties.getProperty(key));
+            }
+        }
+        
+        Map<String, String> requestMap = new HashMap<String, String>();
+        Enumeration<String> names = request.getParameterNames();
+        while (names.hasMoreElements()) {
+            String name = names.nextElement();
+            requestMap.put(name, request.getParameter(name));
+        }
+        
+        context.put( "server", request.getServerName());
+        context.put( "port", request.getServerPort());
+        context.put( "protocol", request.getScheme());
+        
+        context.put("request", request);
+        context.put("requestParams", requestMap);
```

Comment:
Excellent.  FYI for others, what the app launcher (in the JasigWidgetsPortlet project) had as 'request' is now 'requestParams'.  When app launcher converts it will need to have any SpEL URLs changed accordingly.  The confusion of backwards-compatibility is not worth it in this instance IMHO so I had suggested this change (using request for the request and requestParams for the request parameters) as it is simply more clear on intent.